packages/cache-inmemory/src/services/__tests__/inmemory-cache.js

```diff
@@ -0,0 +1,87 @@
+import { InMemoryCacheService } from "../index"
+
+jest.setTimeout(40000)
+
+describe("InMemoryCacheService", () => {
+  let inMemoryCache
+  beforeAll(() => {
+    jest.resetAllMocks()
+  })
+
+  it("Creates an `InMemoryCacheService`", () => {
+    inMemoryCache = new InMemoryCacheService(
+      {},
+      { defaultTTL: 10 },
+      { resources: "shared" }
+    )
+  })
+
+  it("Throws on isolated module declaration", () => {
+    try {
+      inMemoryCache = new InMemoryCacheService(
+        {},
+        {},
+        { resources: "isolated" }
+      )
+    } catch (error) {
+      expect(error.message).toEqual(
+        "At the moment this module can only be used with shared resources"
+      )
+    }
+  })
+
+  it("Stores and retrieves data", async () => {
+    inMemoryCache = new InMemoryCacheService({}, {}, { resources: "shared" })
+
+    await inMemoryCache.set("cache-key", { data: "value" })
+
+    expect(await inMemoryCache.get("cache-key")).toEqual({ data: "value" })
+  })
+
+  it("Invalidates data", async () => {
+    inMemoryCache = new InMemoryCacheService({}, {}, { resources: "shared" })
+
+    await inMemoryCache.set("cache-key", { data: "value" })
+
+    await inMemoryCache.invalidate("cache-key")
+
+    expect(await inMemoryCache.get("cache-key")).toEqual(null)
+  })
+
+  it("Removes data after TTL", async () => {
+    inMemoryCache = new InMemoryCacheService({}, {}, { resources: "shared" })
+
+    await inMemoryCache.set("cache-key", { data: "value" }, 2)
+    expect(await inMemoryCache.get("cache-key")).toEqual({ data: "value" })
+
+    await new Promise((res) => setTimeout(res, 3000))
+
+    expect(await inMemoryCache.get("cache-key")).toEqual(null)
+  })
+
+  it("Removes data after default TTL if TTL params isn't passed", async () => {
```

Comment:
question: It seams to be the same as the previous one since none of them passes a TTL no?

Reply:
in the previous case TTL is passed on L:54